Source/WebCore/html/HTMLFormControlElement.cpp

```diff
@@ -459,7 +459,7 @@ void HTMLFormControlElement::handleInvokeAction()
     invokee->dispatchEvent(event);
 
     if (!event->defaultPrevented())
-        invokee->handleInvokeInternal(action);
+        invokee->handleInvokeInternal(this, action);
```

Comment:
Why does this code call `invokeAction` **before** dispatching the event? Do we have test cases that cover what happens if the event handler changes the `invokeaction` value? We should. This is the kind of thing that can lead to a non-obvious source of incompatibility between different web browsers when people write strange code.

Reply:
It has to call invokeAction before dispatching the event because in a follow up it will only fire the event depending on if the action name meets certain restrictions. It's a good point that we should have a test to ensure that changing the attribute during the event doesn't change what happens.